nixos/modules/programs/way-shell.nix

```diff
@@ -0,0 +1,22 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.way-shell;
+in
+{
+  options.programs.way-shell = {
+    enable = lib.mkEnableOption "way-shell, a GNOME-like shell for Wayland compositors";
+
+    package = lib.mkPackageOption pkgs "way-shell" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    services.upower.enable = true;
```

Comment:
Is this module a work-in-progress?

Reply:
Not quite. From the way-shell [README](https://github.com/ldelossa/way-shell/blob/main/README.md):
```
Way-Shell expects a Gnome-like environment to be available. This means DBus must be running and the following services must be available:

- Logind
- NetworkManager
- WirePlumber/Pipewire
- PowerProfiles Daemon
- UPower
```

uPower is the only service I remember to be unavailable by default, but I could explicitly enable all of them if you prefer.